File: TelerikUIExamplesInCSharp/Examples/SideDrawer/PositionsExample/SideDrawerPositionsModalController.cs
using System;

using ObjCRuntime;
using Foundation;
using UIKit;
using CoreAnimation;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	public class SideDrawerPositionsModalController : SideDrawerTransitionsModalController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			this.SideDrawer.Transition = TKSideDrawerTransitionType.Reveal;
			this.SideDrawer.Fill = new TKSolidFill (UIColor.Gray);
		}

		public override void AddButtons ()
		{
			this.CreateButton ("Left", this, new Selector ("LeftSideDrawer"), new CGPoint (15, 80));
			this.CreateButton ("Right", this, new Selector ("RightSideDrawer"), new CGPoint (15, 130));
			this.CreateButton ("Top", this, new Selector ("TopSideDrawer"), new CGPoint (15, 180));
			this.CreateButton ("Bottom", this, new Selector ("BottomSideDrawer"), new CGPoint (15, 230));
		}

		[Export ("LeftSideDrawer")]
		private void LeftSideDrawer()
		{
			this.SideDrawer.Position = TKSideDrawerPosition.Left;
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			this.SideDrawer.Show ();
		}

		[Export ("RightSideDrawer")]
		private void RightSideDrawer()
		{
			this.SideDrawer.Position = TKSideDrawerPosition.Right;
			this.SideDrawer.HeaderView = new SideDrawerHeader (true, this, new Selector ("DismissSideDrawer"));
			this.SideDrawer.Show ();
		}

		[Export ("TopSideDrawer")]
		private void TopSideDrawer()
		{
			this.SideDrawer.Position = TKSideDrawerPosition.Top;
			this.SideDrawer.HeaderView = new SideDrawerHeader (false, null, null);
			TKSideDrawerTableView table = (TKSideDrawerTableView)this.SideDrawer.Content;
			table.TableView.SetContentOffset (CGPoint.Empty, false);
			this.SideDrawer.AllowScroll = true;
			this.SideDrawer.Show ();
		}

		[Export ("BottomSideDrawer")]
		private void BottomSideDrawer() 
		{
			this.SideDrawer.Position = TKSideDrawerPosition.Bottom;
			this.SideDrawer.HeaderView = new SideDrawerHeader (true, this, new Selector ("DismissSideDrawer"));
			TKSideDrawerTableView table = (TKSideDrawerTableView)this.SideDrawer.Content;
			table.TableView.SetContentOffset (CGPoint.Empty, false);
			this.SideDrawer.AllowScroll = true;
			this.SideDrawer.Show ();
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/SideDrawerPositionsModalController.swift
//
//  SideDrawerPositionsModalController.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class SideDrawerPositionsModalController: SideDrawerGettingStartedModalController, TKSideDrawerDelegate {
   
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.createButton("Left", target: self, selector: "leftSideDrawer", origin: CGPointMake(15, 80))
        self.createButton("Right", target: self, selector: "rightSideDrawer", origin: CGPointMake(15, 130))
        self.createButton("Top", target: self, selector: "topSideDrawer", origin: CGPointMake(15, 180))
        self.createButton("Bottom", target: self, selector: "bottomSideDrawer", origin: CGPointMake(15, 230))
        
        let sideDrawer = self.sideDrawer
        sideDrawer.transition = TKSideDrawerTransitionType.Reveal
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        
    }
    
    func createButton(title: String, target: AnyObject, selector: Selector, origin: CGPoint) {
        let titleString: NSString = title as NSString
        let titleSize = titleString.sizeWithAttributes([NSFontAttributeName : UIFont.systemFontOfSize(18.0)])
        let button = UIButton(frame: CGRectMake(origin.x, origin.y, titleSize.width * 1.5, 44))
        button.titleLabel?.font = UIFont.systemFontOfSize(14)
        button.layer.borderWidth = 1.0
        button.layer.borderColor = UIColor.whiteColor().CGColor
        button.layer.cornerRadius = 3.0
        button.setTitle(title, forState: UIControlState.Normal)
        button.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
        button.addTarget(self, action: selector, forControlEvents: UIControlEvents.TouchUpInside)
        self.view .addSubview(button)
    }
    
    func leftSideDrawer() {
        let sideDrawer = self.sideDrawer
        sideDrawer.position = TKSideDrawerPosition.Left
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func rightSideDrawer() {
        let sideDrawer = self.sideDrawer
        sideDrawer.position = TKSideDrawerPosition.Right
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: Selector("dismissSideDrawer"))
        self.showSideDrawer()
    }
    
    func topSideDrawer() {
        let sideDrawer = self.sideDrawer
        sideDrawer.position = TKSideDrawerPosition.Top
        (sideDrawer.content as TKSideDrawerTableView).tableView .setContentOffset(CGPointZero, animated: false)
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        sideDrawer.allowScroll = true
        self.showSideDrawer()
    }
    
    func bottomSideDrawer() {
        let sideDrawer = self.sideDrawer
        sideDrawer.position = TKSideDrawerPosition.Bottom
        (sideDrawer.content as TKSideDrawerTableView).tableView .setContentOffset(CGPointZero, animated: false)
        sideDrawer.allowScroll = true
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: Selector("dismissSideDrawer"))
        self.showSideDrawer()
    }
}
